qcodes/instrument_drivers/Lakeshore/lakeshore_base.py

```diff
@@ -0,0 +1,302 @@
+from typing import Dict, ClassVar
+import logging
+from collections import OrderedDict
+import time
+from bisect import bisect
+
+from qcodes import VisaInstrument, InstrumentChannel, ChannelList
+from qcodes.instrument.parameter import Parameter
+from qcodes.utils import validators
+
+log = logging.getLogger(__name__)
+
+class GroupParameter(Parameter):
+    def __init__(self, name, instrument, **kwargs):
+        self.group = None
+        super().__init__(name, instrument, **kwargs)
+
+    def set_raw(self, value, **kwargs):  # pylint: disable=E0202
+        self.group.set(self, value)
+
+    def get_raw(self, result=None):  # pylint: disable=E0202
+        if not result:
+            self.group.update()
+            return self.raw_value
+        else:
+            return result
+
+
+class Group():
+    def __init__(self, parameters, set_cmd=None, get_cmd=None,
+                 get_parser=None, separator=',', types=None) -> None:
+        self.parameters = OrderedDict((p.name, p) for p in parameters)
+        self.instrument = parameters[0].root_instrument
+        for p in parameters:
+            p.group = self
+        self.set_cmd = set_cmd
+        self.get_cmd = get_cmd
+        if get_parser:
+            self.get_parser = get_parser
+        else:
+            self.get_parser = self._separator_parser(separator, types)
+
+    def _separator_parser(self, separator, types):
+        def parser(ret_str):
+            keys = self.parameters.keys()
+            values = ret_str.split(separator)
+            return dict(zip(keys, values))
+        return parser
+
+    def set(self, set_parameter, value):
+        if any((p.get_latest() is None) for p in self.parameters.values()):
+            self.update()
+        calling_dict = {name: p.raw_value
+                        for name, p in self.parameters.items()}
+        calling_dict[set_parameter.name] = value
+        command_str = self.set_cmd.format(**calling_dict)
```

Comment:
this means that the `Group` only supports string commands, right? Shall we generalize it to callables as well?

Reply:
maybe later, but not in this PR